### ISSUE TITLE
Rule `no-tabs` being triggered in `lineinfile` module calls

### ISSUE THREAD

**Issue by hegyibalint** [ansible/ansible-lint#3010]
Issue detected on `6.12.2.dev0`
Issue verified on HEAD (`6.12.2.dev15`)

##### Summary

According to [no-tab's documentation](https://ansible-lint.readthedocs.io/rules/no-tabs/):
> This rule does not trigger alerts for tab characters in the ansible.builtin.lineinfile module.

 - In playbooks, this is the behavior.
 - In roles, the rule will trigger nonetheless.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

OS: Arch Linux

Ansible-lint version:
```
ansible-lint --version
ansible-lint 6.12.2.dev0 using ansible 2.14.2
You are using a pre-release version of ansible-lint.
```
(As mentioned, it also happens on `master`)

- ansible installation method: `pacman -S ansible` (i.e. OS package)
- ansible-lint installation method: `pacman -S ansible-lint` (i.e. OS package)

##### STEPS TO REPRODUCE

https://github.com/hegyibalint/ansible-lint-no-tab-fail-reproducer

##### Desired Behavior

The rule should not be triggered in `lineinfile` module calls

**Comment by ssbarnea** [comment]
My (personal) opinion is that if tabs are required in YAML strings, they should always be escaped as `\t`, as otherwise it is too likely for them to not be noticed or to accidentally be changed to spaces.

Unless that is not really possible to achieve, I would say this this bug is likely to be sorted by just improving the documentation on this rule.
